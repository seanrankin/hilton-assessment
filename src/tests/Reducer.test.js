import {
  SET_ROOM_ENABLED,
  UPDATE_ADULTS,
  UPDATE_CHILDREN,
} from '../actions/reservations';
import reducer, { initialState } from '../reducers/reservations';

describe('Reservations reducer', () => {
  it('should return the initial state', () => {
    expect(reducer(undefined, {})).toEqual(initialState);
    expect(initialState).toMatchSnapshot();
  });

  it('should handle SET_ROOM_ENABLED', () => {
    const successAction = {
      type: SET_ROOM_ENABLED,
      id: 3,
    };

    const successObject = {
      room1: {
        adults: 1,
        children: 0,
        checked: true,
      },
      room2: {
        adults: 1,
        children: 0,
        checked: false,
      },
      room3: {
        adults: 1,
        children: 0,
        checked: true,
      },
      room4: {
        adults: 1,
        children: 0,
        checked: false,
      },
    };

    expect(reducer(initialState, successAction)).toEqual(successObject);
  });

  it('should handle UPDATE_ADULTS', () => {
    const successAction = {
      type: UPDATE_ADULTS,
      id: 1,
      adults: 2,
    };

    const successObject = {
      room1: {
        adults: 2,
        children: 0,
        checked: true,
      },
      room2: {
        adults: 1,
        children: 0,
        checked: false,
      },
      room3: {
        adults: 1,
        children: 0,
        checked: false,
      },
      room4: {
        adults: 1,
        children: 0,
        checked: false,
      },
    };

    expect(reducer(initialState, successAction)).toEqual(successObject);
  });

  it('should handle UPDATE_CHILDREN', () => {
    const successAction = {
      type: UPDATE_CHILDREN,
      id: 1,
      children: 2,
    };

    const successObject = {
      room1: {
        adults: 1,
        children: 2,
        checked: true,
      },
      room2: {
        adults: 1,
        children: 0,
        checked: false,
      },
      room3: {
        adults: 1,
        children: 0,
        checked: false,
      },
      room4: {
        adults: 1,
        children: 0,
        checked: false,
      },
    };

    expect(reducer(initialState, successAction)).toEqual(successObject);
  });
});
