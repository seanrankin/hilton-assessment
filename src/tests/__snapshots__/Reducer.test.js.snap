// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reservations reducer should return the initial state 1`] = `
Object {
  "room1": Object {
    "adults": 1,
    "checked": true,
    "children": 0,
  },
  "room2": Object {
    "adults": 1,
    "checked": false,
    "children": 0,
  },
  "room3": Object {
    "adults": 1,
    "checked": false,
    "children": 0,
  },
  "room4": Object {
    "adults": 1,
    "checked": false,
    "children": 0,
  },
}
`;
